app/code/Magento/CatalogWidget/Model/Rule/Condition/Product.php

```diff
@@ -97,6 +104,13 @@ function ($attribute) {
 
         $attributes = [];
         foreach ($productAttributes as $attribute) {
```

Comment:
As the method loadAttributeOptions() is being functionally changed, the testsuite test for the method should be updated to take into consideration the attribute can now affect the output.